File: 4기 코딩테스트_220615/4기 - 2nd 코딩테스트/AsciiToString/AsciiToString_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.supereasy.AsciiToString.AsciiToString import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        arr = [71, 111, 111, 103, 108, 101]

        # when
        result = solution(arr)

        # then
        self.assertEqual('Google', result)

    def test_2(self):
        # given
        arr = [65, 108, 71, 111, 82, 105, 116, 72, 109]

        # when
        result = solution(arr)

        # then
        self.assertEqual('AlGoRitHm', result)

    def test_3(self):
        # given
        arr = []

        # when
        result = solution(arr)

        # then
        self.assertEqual('', result)

    def test_4(self):
        # given
        arr = [65]

        # when
        result = solution(arr)

        # then
        self.assertEqual('A', result)

    def test_5(self):
        # given
        arr = [122]

        # when
        result = solution(arr)

        # then
        self.assertEqual('z', result)

    def test_6(self):
        # given
        arr = [119, 111, 114, 107]

        # when
        result = solution(arr)

        # then
        self.assertEqual('work', result)


if __name__ == '__main__':
    unittest.main()

File: 4기 코딩테스트_220615/4기 - 2nd 코딩테스트/AsciiToString/AsciiToString_test.js
import assert from 'assert'
import solution from './AsciiToString.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const arr = [71, 111, 111, 103, 108, 101]

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual('Google', result)
    done();
  });

  it('test 2', function (done) {
    // given
    const arr = [65, 108, 71, 111, 82, 105, 116, 72, 109]

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual('AlGoRitHm', result)
    done();
  });

  it('test 3', function (done) {
    // given
    const arr = []

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual('', result)
    done();
  });

  it('test 4', function (done) {
    // given
    const arr = [65]

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual('A', result)
    done();
  });

  it('test 5', function (done) {
    // given
    const arr = [122]

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual('z', result)
    done();
  });

  it('test 6', function (done) {
    // given
    const arr = [119, 111, 114, 107]

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual('work', result)
    done();
  });
})
